File: .conver/ui/screens/ApprovalScreen.kt
package com.example.ui.screens

import androidx.compose.animation.*
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.model.ApprovalRequest
import com.example.ui.theme.TealTertiary
import com.example.ui.viewmodel.AppViewModel
import java.text.SimpleDateFormat
import java.util.*

@Composable
fun ApprovalScreen(
    viewModel: AppViewModel,
    modifier: Modifier = Modifier
) {
    val approvalRequests by viewModel.approvalRequests.collectAsState()

    val pendingRequests = remember(approvalRequests) {
        approvalRequests.filter { it.status == "Menunggu" }
    }

    val historyRequests = remember(approvalRequests) {
        approvalRequests.filter { it.status != "Menunggu" }
    }

    var selectedTab by remember { mutableStateOf(0) } // 0 = Menunggu, 1 = Riwayat

    Column(
        modifier = modifier
            .fillMaxSize()
            .padding(horizontal = 16.dp)
    ) {
        // Upper stats banner
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 12.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Card(
                modifier = Modifier
                    .weight(1f)
                    .height(64.dp),
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.primary.copy(alpha = 0.08f)),
                border = BorderStroke(1.dp, MaterialTheme.colorScheme.primary.copy(alpha = 0.2f))
            ) {
                Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text("${pendingRequests.size}", style = MaterialTheme.typography.titleLarge, fontWeight = FontWeight.Bold, color = MaterialTheme.colorScheme.primary)
                        Text("Menunggu Tindakan", style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                    }
                }
            }

            Card(
                modifier = Modifier
                    .weight(1f)
                    .height(64.dp),
                colors = CardDefaults.cardColors(containerColor = TealTertiary.copy(alpha = 0.08f)),
                border = BorderStroke(1.dp, TealTertiary.copy(alpha = 0.2f))
            ) {
                Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text("${historyRequests.count { it.status == "Disetujui" }}", style = MaterialTheme.typography.titleLarge, fontWeight = FontWeight.Bold, color = TealTertiary)
                        Text("Total Disetujui", style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
                    }
                }
            }
        }

        // Selection Tab Row
        TabRow(
            selectedTabIndex = selectedTab,
            containerColor = Color.Transparent,
            modifier = Modifier.padding(bottom = 12.dp)
        ) {
            Tab(
                selected = selectedTab == 0,
                onClick = { selectedTab = 0 },
                text = { Text("Menunggu (${pendingRequests.size})", fontWeight = FontWeight.Bold) },
                modifier = Modifier.testTag("pending_approvals_tab")
            )
            Tab(
                selected = selectedTab == 1,
                onClick = { selectedTab = 1 },
                text = { Text("Riwayat (${historyRequests.size})", fontWeight = FontWeight.Bold) },
                modifier = Modifier.testTag("history_approvals_tab")
            )
        }

        // Selected requests lists
        val displayedRequests = if (selectedTab == 0) pendingRequests else historyRequests

        if (displayedRequests.isEmpty()) {
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .weight(1f),
                contentAlignment = Alignment.Center
            ) {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Icon(
                        imageVector = if (selectedTab == 0) Icons.Default.DoneAll else Icons.Default.History,
                        contentDescription = "Empty",
                        tint = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.size(64.dp)
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = if (selectedTab == 0) "Selesai! Tidak ada permintaan tertunda" else "Riwayat permintaan masih kosong",
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        style = MaterialTheme.typography.bodyMedium
                    )
                }
            }
        } else {
            LazyColumn(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(12.dp),
                contentPadding = PaddingValues(bottom = 24.dp)
            ) {
                items(displayedRequests, key = { it.id }) { request ->
                    ApprovalRowItem(
                        request = request,
                        isHistory = selectedTab == 1,
                        onApprove = { viewModel.approveRequest(request.id) },
                        onReject = { viewModel.rejectRequest(request.id) }
                    )
                }
            }
        }
    }
}

@Composable
fun ApprovalRowItem(
    request: ApprovalRequest,
    isHistory: Boolean,
    onApprove: () -> Unit,
    onReject: () -> Unit
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .shadow(1.dp, RoundedCornerShape(16.dp)),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        border = BorderStroke(1.dp, MaterialTheme.colorScheme.surfaceVariant)
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            // Request Header
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    modifier = Modifier.weight(1f)
                ) {
                    Box(
                        modifier = Modifier
                            .size(36.dp)
                            .background(
                                color = when (request.title) {
                                    "Pendaftaran Seller Baru" -> MaterialTheme.colorScheme.primary.copy(alpha = 0.15f)
                                    "Permintaan Buka Blokir" -> MaterialTheme.colorScheme.error.copy(alpha = 0.15f)
                                    else -> TealTertiary.copy(alpha = 0.15f)
                                },
                                shape = CircleShape
                            ),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = when (request.title) {
                                "Pendaftaran Seller Baru" -> Icons.Default.PersonAddAlt
                                "Permintaan Buka Blokir" -> Icons.Default.LockOpen
                                else -> Icons.Default.CloudQueue
                            },
                            contentDescription = null,
                            tint = when (request.title) {
                                "Pendaftaran Seller Baru" -> MaterialTheme.colorScheme.primary
                                "Permintaan Buka Blokir" -> MaterialTheme.colorScheme.error
                                else -> TealTertiary
                            },
                            modifier = Modifier.size(18.dp)
                        )
                    }
                    Spacer(modifier = Modifier.width(12.dp))
                    Column {
                        Text(
                            text = request.title,
                            style = MaterialTheme.typography.titleSmall,
                            fontWeight = FontWeight.Bold,
                            color = MaterialTheme.colorScheme.onSurface
                        )
                        Text(
                            text = "Pemohon: ${request.requesterName}",
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }

                val sdf = SimpleDateFormat("HH:mm dd MMM", Locale.getDefault())
                Text(
                    text = sdf.format(Date(request.timestamp)),
                    style = MaterialTheme.typography.labelSmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.8f),
                    fontSize = 10.sp
                )
            }

            Spacer(modifier = Modifier.height(12.dp))

            // Details section
            Text(
                text = request.details,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurface,
                maxLines = 3,
                overflow = TextOverflow.Ellipsis
            )

            Spacer(modifier = Modifier.height(16.dp))

            // Action section or history indicator status tag
            if (!isHistory) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(12.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    OutlinedButton(
                        onClick = onReject,
                        shape = RoundedCornerShape(10.dp),
                        border = BorderStroke(1.dp, MaterialTheme.colorScheme.error),
                        colors = ButtonDefaults.outlinedButtonColors(contentColor = MaterialTheme.colorScheme.error),
                        modifier = Modifier
                            .weight(1f)
                            .height(44.dp)
                            .testTag("reject_button_${request.id}")
                    ) {
                        Icon(Icons.Default.Close, contentDescription = null, modifier = Modifier.size(16.dp))
                        Spacer(modifier = Modifier.width(4.dp))
                        Text("Tolak", fontWeight = FontWeight.Bold, fontSize = 13.sp)
                    }

                    Button(
                        onClick = onApprove,
                        shape = RoundedCornerShape(10.dp),
                        colors = ButtonDefaults.buttonColors(containerColor = TealTertiary),
                        modifier = Modifier
                            .weight(1f)
                            .height(44.dp)
                            .testTag("approve_button_${request.id}")
                    ) {
                        Icon(Icons.Default.Check, contentDescription = null, modifier = Modifier.size(16.dp))
                        Spacer(modifier = Modifier.width(4.dp))
                        Text("Setujui", fontWeight = FontWeight.Bold, fontSize = 13.sp, color = Color.White)
                    }
                }
            } else {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.End,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    val isApproved = request.status == "Disetujui"
                    Box(
                        modifier = Modifier
                            .clip(RoundedCornerShape(8.dp))
                            .background(
                                color = if (isApproved) TealTertiary.copy(alpha = 0.15f) else MaterialTheme.colorScheme.error.copy(alpha = 0.15f)
                            )
                            .padding(horizontal = 12.dp, vertical = 6.dp)
                    ) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                imageVector = if (isApproved) Icons.Default.CheckCircle else Icons.Default.Cancel,
                                contentDescription = null,
                                tint = if (isApproved) TealTertiary else MaterialTheme.colorScheme.error,
                                modifier = Modifier.size(14.dp)
                            )
                            Spacer(modifier = Modifier.width(4.dp))
                            Text(
                                text = request.status.uppercase(),
                                style = MaterialTheme.typography.labelSmall,
                                fontWeight = FontWeight.Bold,
                                color = if (isApproved) TealTertiary else MaterialTheme.colorScheme.error
                            )
                        }
                    }
                }
            }
        }
    }
}

File: lib/screens/approval_screen.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import 'package:provider/provider.dart';
import '../models/approval_request.dart';
import '../viewmodels/app_view_model.dart';


class ApprovalScreen extends StatefulWidget {
  const ApprovalScreen({super.key});

  @override
  State<ApprovalScreen> createState() => _ApprovalScreenState();
}

class _ApprovalScreenState extends State<ApprovalScreen> with SingleTickerProviderStateMixin {
  late TabController _tabController;

  @override
  void initState() {
    super.initState();
    _tabController = TabController(length: 2, vsync: this);
  }

  @override
  void dispose() {
    _tabController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final viewModel = Provider.of<AppViewModel>(context);
    final allRequests = viewModel.approvalRequests;
    final pendingRequests = allRequests.where((r) => r.status == "Menunggu").toList();
    final historyRequests = allRequests.where((r) => r.status != "Menunggu").toList();

    return Scaffold(
      backgroundColor: Theme.of(context).scaffoldBackgroundColor,
      body: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 16.0),
        child: Column(
          children: [
            const SizedBox(height: kToolbarHeight - 20),
            _StatsBanners(pendingCount: pendingRequests.length, approvedCount: historyRequests.where((r)=> r.status == "Disetujui").length),
            const SizedBox(height: 12),
            _buildTabs(pendingRequests.length, historyRequests.length),
             const SizedBox(height: 12),
            Expanded(
              child: TabBarView(
                controller: _tabController,
                children: [
                  _RequestList(requests: pendingRequests, isHistory: false),
                  _RequestList(requests: historyRequests, isHistory: true),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildTabs(int pendingCount, int historyCount) {
    return TabBar(
      controller: _tabController,
      tabs: [
        Tab(text: "Menunggu ($pendingCount)"),
        Tab(text: "Riwayat ($historyCount)"),
      ],
       labelStyle: const TextStyle(fontWeight: FontWeight.bold),
       unselectedLabelStyle: const TextStyle(fontWeight: FontWeight.normal),
    );
  }
}

class _StatsBanners extends StatelessWidget {
  final int pendingCount;
  final int approvedCount;

  const _StatsBanners({required this.pendingCount, required this.approvedCount});

  @override
  Widget build(BuildContext context) {
    return Row(
      children: [
        Expanded(child: _StatCard(count: pendingCount, label: "Menunggu Tindakan", color: Theme.of(context).colorScheme.primary)),
        const SizedBox(width: 12),
        Expanded(child: _StatCard(count: approvedCount, label: "Total Disetujui", color: const Color(0xFF26A69A))),
      ],
    );
  }
}

class _StatCard extends StatelessWidget {
  final int count;
  final String label;
  final Color color;

  const _StatCard({required this.count, required this.label, required this.color});

  @override
  Widget build(BuildContext context) {
    return Card(
      color: color.withAlpha(8),
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(8),
        side: BorderSide(color: color.withAlpha(2)),
      ),
      elevation: 0,
      child: SizedBox(
        height: 64,
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Text(count.toString(), style: Theme.of(context).textTheme.titleLarge?.copyWith(fontWeight: FontWeight.bold, color: color)),
            Text(label, style: Theme.of(context).textTheme.labelSmall?.copyWith(color: Theme.of(context).colorScheme.onSurfaceVariant)),
          ],
        ),
      ),
    );
  }
}

class _RequestList extends StatelessWidget {
  final List<ApprovalRequest> requests;
  final bool isHistory;

  const _RequestList({required this.requests, required this.isHistory});

  @override
  Widget build(BuildContext context) {
    if (requests.isEmpty) {
      final icon = isHistory ? Icons.history : Icons.done_all;
      final message = isHistory ? "Riwayat permintaan masih kosong" : "Selesai! Tidak ada permintaan tertunda";
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Icon(icon, size: 64, color: Theme.of(context).colorScheme.onSurfaceVariant),
            const SizedBox(height: 8),
            Text(message, style: TextStyle(color: Theme.of(context).colorScheme.onSurfaceVariant)),
          ],
        ),
      );
    }

    return ListView.separated(
      padding: const EdgeInsets.only(bottom: 24),
      itemCount: requests.length,
      itemBuilder: (context, index) => _ApprovalRowItem(request: requests[index], isHistory: isHistory),
      separatorBuilder: (context, index) => const SizedBox(height: 12),
    );
  }
}

class _ApprovalRowItem extends StatelessWidget {
  final ApprovalRequest request;
  final bool isHistory;

  const _ApprovalRowItem({required this.request, required this.isHistory});

  @override
  Widget build(BuildContext context) {
    final viewModel = Provider.of<AppViewModel>(context, listen: false);
    final colorScheme = Theme.of(context).colorScheme;
    final tealColor = const Color(0xFF26A69A);

    IconData iconData;
    Color iconColor;

    switch (request.title) {
      case "Pendaftaran Seller Baru":
        iconData = Icons.person_add_alt;
        iconColor = colorScheme.primary;
        break;
      case "Permintaan Buka Blokir":
        iconData = Icons.lock_open;
        iconColor = colorScheme.error;
        break;
      default:
        iconData = Icons.cloud_queue;
        iconColor = tealColor;
    }

    return Card(
        elevation: 1,
        shadowColor: Colors.black.withAlpha(1),
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(16),
          side: BorderSide(color: colorScheme.surfaceContainerHighest),
        ),
        child: Padding(
          padding: const EdgeInsets.all(16.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Row(
                 mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Row(children: [ 
                     CircleAvatar(
                      radius: 18,
                      backgroundColor: iconColor.withAlpha(15),
                      child: Icon(iconData, color: iconColor, size: 18),
                    ),
                    const SizedBox(width: 12),
                    Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(request.title, style: Theme.of(context).textTheme.titleSmall?.copyWith(fontWeight: FontWeight.bold)),
                        Text("Pemohon: ${request.requesterName}", style: Theme.of(context).textTheme.bodySmall),
                      ],
                    ),   
                  ],),
                   Text(
                      DateFormat("HH:mm dd MMM").format(request.timestamp.toLocal()),
                      style: Theme.of(context).textTheme.labelSmall?.copyWith(fontSize: 10, color: colorScheme.onSurfaceVariant),
                    ),
                 
                ],
              ),
              const SizedBox(height: 12),
              Text(request.details, maxLines: 3, overflow: TextOverflow.ellipsis),
              const SizedBox(height: 16),
              if (isHistory)
                _buildHistoryStatus(context, tealColor, colorScheme)
              else
                _buildActionButtons(context, viewModel, tealColor, colorScheme),
            ],
          ),
        ));
  }

  Widget _buildActionButtons(BuildContext context, AppViewModel viewModel, Color tealColor, ColorScheme colorScheme) {
    return Row(
      children: [
        Expanded(
          child: OutlinedButton.icon(
            onPressed: () => viewModel.rejectRequest(request.id),
            icon: const Icon(Icons.close, size: 16), 
            label: const Text("Tolak"),
            style: OutlinedButton.styleFrom(
              foregroundColor: colorScheme.error,
              side: BorderSide(color: colorScheme.error),
            ),
          ),
        ),
        const SizedBox(width: 12),
        Expanded(
          child: ElevatedButton.icon(
            onPressed: () => viewModel.approveRequest(request.id),
            icon: const Icon(Icons.check, size: 16), 
            label: const Text("Setujui"),
            style: ElevatedButton.styleFrom(
              backgroundColor: tealColor,
              foregroundColor: Colors.white,
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildHistoryStatus(BuildContext context, Color tealColor, ColorScheme colorScheme) {
    final isApproved = request.status == "Disetujui";
    final statusColor = isApproved ? tealColor : colorScheme.error;
    final statusIcon = isApproved ? Icons.check_circle : Icons.cancel;

    return Align(
      alignment: Alignment.centerRight,
      child: Container(
        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
        decoration: BoxDecoration(
          color: statusColor.withAlpha(15),
          borderRadius: BorderRadius.circular(8),
        ),
        child: Row(
          mainAxisSize: MainAxisSize.min,
          children: [
            Icon(statusIcon, color: statusColor, size: 14),
            const SizedBox(width: 4),
            Text(
              request.status.toUpperCase(),
              style: Theme.of(context).textTheme.labelSmall?.copyWith(fontWeight: FontWeight.bold, color: statusColor),
            ),
          ],
        ),
      ),
    );
  }
}
